bigchaindb/backend/mongodb/connection.py

```diff
@@ -63,8 +73,35 @@ def run(self, query):
         except pymongo.errors.OperationFailure as exc:
             raise OperationError from exc
 
+    def _secure_connect(self):
+        """
+        _secure_connect() checks if the implied ssl=True parameter is not False,
+        that is, if the configuration specifies ca_cert, certfile, keyfile and
+        crl, then the ssl parameter needs to be set to True.
+        If not, it raises a ConfigurationError.
+
+        Raises:
+            :exc:`~ConfigurationError`: If ssl=false in the specified configuration.
+        """
+
+        logger.info('Connecting to MongoDB over TLS/SSL...')
+        client = pymongo.MongoClient(self.host,
+                                     self.port,
+                                     replicaset=self.replicaset,
+                                     serverselectiontimeoutms=self.connection_timeout,
+                                     ssl=self.ssl,
+                                     ssl_ca_certs=self.ca_cert,
+                                     ssl_certfile=self.certfile,
+                                     ssl_keyfile=self.keyfile,
+                                     ssl_pem_passphrase=self.keyfile_passphrase,
+                                     ssl_crlfile=self.crlfile,
+                                     ssl_cert_reqs=CERT_REQUIRED)
+        return client
+
     def _connect(self):
         """Try to connect to the database.
+        This method reads the configuration options and calls
+        _secure_connect(), if required
 
         Raises:
```

Comment:
The `AuthenticationError` was not introduced in this pr but could you add it here also?